server/src/domain/search/search.service.ts

```diff
@@ -122,4 +126,52 @@ export class SearchService {
     userIds.push(...partnersIds);
     return userIds;
   }
+
+  async getSearchSuggestions(auth: AuthDto, dto: SearchSuggestionRequestDto): Promise<SearchSuggestionResponseDto> {
```

Comment:
Ah, I am thinking we are not running all of the suggestion in one API call though. So on the filter form, depends on the type of filter you are wanting to use, it will run the corresponding services to get the suggestion